### DISCUSSION TITLE
receive webhook from payment gateway

### DISCUSSION THREAD

**Discussion by felnanuke2** [supabase/supabase#3462] [Questions] [answered]
my application needs to receive webhook from the paid marketplace when a customer pays.  the webhook is a post with some arguments in the body, but that's not a problem I created a function to accept the parameters and handle the request, but the request sends in the url some path parameters that aren't accepted by the api, apparently.  in this case I tell the gateway to notify the corresponding url and pass the url in the apikey to authorize the request, but the gateway adds some extra arguments to the url.  I tested it with postman and when I pass some path parameters that the api doesn't expect it returns an error asking me to use something like .eq and saying that the column doesn't exist.  is there any way to do this or will i have to create an api to do this integration??

**Comment by steve-chavez** [comment, marked as answer]
> but the request sends in the url some path parameters

Here you mean the webhook sends query parameters like `?some_param=yy&other_param=xx` right? Are those query parameters well-defined?(the proxy doesn't add/remove parameters dinamically? Do you know the names of these parameters?)

> that aren't accepted by the api, apparently

Yes, that's true. Recently we added the capability that certain functions - those declared with a single unnamed json parameter - can receive a json with any number of keys/values, but it only works for the POST body. In theory, it could be extended to also handle other parameters beside the unnamed json param but that would only work if the URI query parameters are well-defined.

**Edit**: perhaps we can have a second unnamed json parameter to a function to mean all extra query string parameters go here. Would still need to check if the webhook doesn't conflict with the API(PostgREST) reserved parameters like `?select&order&columns`.

**Reply by felnanuke2** [reply]
the parameters are: ?client_id=something type=something else.  can i solve this by just putting an unnamed json parameter in the function?

**Reply by steve-chavez** [reply]
> can i solve this by just putting an unnamed json parameter in the function?

You can use that now, but as I mentioned it only helps for an unstructured POST body for RPC.

> the parameters are: ?client_id=something type=something

I think it could be a future enhancement for the api to handle unstructured query params plus unstructured POST body. For now, you'd have to create an external backend component for this.

**Comment by felnanuke2** [comment]
ok thanks for the quick reply

**Reply by software-OSG** [reply]
Hello,

How to retrieve data from webhook (e.g. WooCommerce webhook or Stripe webhook) to save in supabase?
